File: src/main/java/club/redux/sunset/lavafishing/LavaFishing.java
package club.redux.sunset.lavafishing;

import club.redux.sunset.lavafishing.util.Reference;
import club.redux.sunset.lavafishing.util.RegistryCollection.*;
import net.minecraftforge.eventbus.api.IEventBus;
import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.fml.javafmlmod.FMLJavaModLoadingContext;

@Mod(Reference.MOD_ID)
public class LavaFishing {

    public LavaFishing() {
        IEventBus eventBus = FMLJavaModLoadingContext.get().getModEventBus();
        BlockCollection.BLOCK_DEFERRED_REGISTER.register(eventBus);
        CreativeModeTabCollection.CREATIVE_MODE_TAB_DEFERRED_REGISTER.register(eventBus);
        BlockEntityTypeCollection.BLOCK_ENTITY_TYPE_DEFERRED_REGISTER.register(eventBus);
        MobEffectCollection.MOB_EFFECT_DEFERRED_REGISTER.register(eventBus);
        EntityTypeCollection.ENTITY_TYPE_DEFERRED_REGISTER.register(eventBus);
        ItemCollection.ITEM_DEFERRED_REGISTER.register(eventBus);
        ParticleTypeCollection.PARTICLE_TYPE_DEFERRED_REGISTER.register(eventBus);
        PotionCollection.POTION_DEFERRED_REGISTER.register(eventBus);
    }
}

File: src/main/kotlin/club/redux/sunset/lavafishing/LavaFishing.kt
package club.redux.sunset.lavafishing

import club.redux.sunset.lavafishing.registry.*
import club.redux.sunset.lavafishing.util.Utils
import club.redux.sunset.lavafishing.util.Utils.resourceLocation
import com.mojang.logging.LogUtils
import net.minecraftforge.fml.common.Mod
import org.slf4j.Logger
import thedarkcolour.kotlinforforge.forge.MOD_BUS

@Mod(BuiltConstants.MOD_ID)
object LavaFishing {
    val Any.logger: Logger by lazy { LogUtils.getLogger() }

    fun resourceLocation(path: String) = BuiltConstants.MOD_ID.resourceLocation(path)

    init {
        Utils.asciiArt()

        listOf(
            ModBlockEntityTypes,
            ModBlocks,
            ModCreativeModeTabs,
            ModEntityTypes,
            ModItems,
            ModItemsAqua,
            ModMobEffects,
            ModParticleTypes,
            ModPotions,
            ModRecipeSerializers,
            ModSoundEvents
        ).forEach { it.attach(MOD_BUS) }
    }
}
